spring-cloud-skipper/src/main/java/org/springframework/cloud/skipper/domain/SpringCloudDeployerApplicationManifestReader.java

```diff
@@ -39,43 +41,55 @@
  * @author Mark Pollack
  * @author Ilayaperumal Gopinathan
  */
-public class SpringCloudDeployerApplicationManifestReader {
+public class SpringCloudDeployerApplicationManifestReader implements SkipperManifestReader {
 
 	private final static Logger logger = LoggerFactory.getLogger(SpringCloudDeployerApplicationManifestReader.class);
 
 	public List<SpringCloudDeployerApplicationManifest> read(String manifest) {
-		assertSupportedKinds(manifest);
-		List<SpringCloudDeployerApplicationManifest> applicationSpecs = new ArrayList<>();
-		YAMLMapper mapper = new YAMLMapper();
-		mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-		try {
-			MappingIterator<SpringCloudDeployerApplicationManifest> it = mapper
-					.readerFor(SpringCloudDeployerApplicationManifest.class).readValues(manifest);
-			while (it.hasNextValue()) {
-				SpringCloudDeployerApplicationManifest appKind = it.next();
-				applicationSpecs.add(appKind);
+		if (assertSupportedKinds(manifest)) {
+			List<SpringCloudDeployerApplicationManifest> applicationSpecs = new ArrayList<>();
+			YAMLMapper mapper = new YAMLMapper();
+			mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+			try {
+				MappingIterator<SpringCloudDeployerApplicationManifest> it = mapper.readerFor(
+						SpringCloudDeployerApplicationManifest.class).readValues(manifest);
+				while (it.hasNextValue()) {
+					SpringCloudDeployerApplicationManifest appKind = it.next();
+					applicationSpecs.add(appKind);
+				}
 			}
+			catch (JsonMappingException e) {
+				logger.error("Can't parse Package's manifest YAML = " + manifest);
+				throw new SkipperException("JsonMappingException - Can't parse Package's manifest YAML = " + manifest,
+						e);
+			}
+			catch (IOException e) {
+				logger.error("Can't parse Package's manifest YAML = " + manifest);
+				throw new SkipperException("IOException - Can't parse Package's manifest YAML = " + manifest, e);
+			}
+			return applicationSpecs;
 		}
-		catch (JsonMappingException e) {
-			logger.error("Can't parse Package's manifest YAML = " + manifest);
-			throw new SkipperException("JsonMappingException - Can't parse Package's manifest YAML = " + manifest, e);
-		}
-		catch (IOException e) {
-			logger.error("Can't parse Package's manifest YAML = " + manifest);
-			throw new SkipperException("IOException - Can't parse Package's manifest YAML = " + manifest, e);
-		}
-		return applicationSpecs;
+		return Collections.emptyList();
 	}
 
-	private void assertSupportedKinds(String manifest) {
+	public boolean assertSupportedKinds(String manifest) {
```

Comment:
If we need these as public methods, should rename as original functionality was more or less asserting manifest. Something like `canSupport` would probably be more appropriate.